### ISSUE TITLE
purl2dwarf: Get DWARFs from a PURL

### ISSUE THREAD

**Issue by pombredanne** [nexB/purldb#260]
Create a PURL a service backed by new SCIO analysis pipelines that takes a PURL for a native binary package as an input, and returns the DWARF debug symbol compilation unit paths when available. This will specifically be worked out for Debian native binary packages.

**Comment by armijnhemel** [comment]
As a package can contain multiple binaries I guess that the result would be a list of "DWARF information per binary"?

**Comment by pombredanne** [comment]
> As a package can contain multiple binaries I guess that the result would be a list of "DWARF information per binary"?

Yes, that's per binary file

**Comment by armijnhemel** [comment]
> > As a package can contain multiple binaries I guess that the result would be a list of "DWARF information per binary"?
> 
> Yes, that's per binary file

What would this look like for a package like coreutils (with ~100 binaries)?

**Comment by pombredanne** [comment]
> What would this look like for a package like coreutils (with ~100 binaries)?

more or less like this:
```YAML
- files:
  - path: foo/bar/find
    dwarf_paths:
      - /home/joe/coreutil/includes/baz.h
      - /home/joe/coreutil/find.c
  - path: foo/bar/ls
    dwarf_paths:
      - /home/joe/coreutil/includes/baz.h
      - /home/joe/coreutil/js.c
```

**Comment by TG1999** [comment]
It should be an add-on pipeline in the data model, it should be used for d2d on elfs. This part is just to collect dwarfs.

**Comment by pombredanne** [comment]
The basis would be provided by the https://github.com/nexb/elf-inspector

**Comment by TG1999** [comment]
Will be closed by https://github.com/nexB/scancode.io/pull/1068

**Comment by pombredanne** [comment]
@TG1999 you should sync up with @keshav-space on this https://github.com/nexB/purldb/issues/260#issuecomment-1997542167 as he has some things in the works for purl2sym

**Comment by pombredanne** [comment]
A good example of a package with DWARFs is https://repo1.maven.org/maven2/com/sun/jna/jna/3.0.9/jna-3.0.9.jar which has the PURL: `pkg:maven/com.sun.jna/jna@3.0.9`

**Comment by pombredanne** [comment]
Related is this issue: https://github.com/nexB/scancode.io/issues/1121

Another example `pkg:generic/debug@1.23?download_url=https://github.com/nexB/elf-inspector/raw/4333e1601229da87fa88961389d7397af6e027c4/tests/data/dwarf_and_elf/analyze.so.debug`

**Comment by JonoYang** [comment]
@TG1999 I've merged #353
